testcafe/AppliancePageTC.js

```diff
@@ -0,0 +1,227 @@
+import {Selector, t} from "testcafe";
+import {waitForReact, ReactSelector} from "testcafe-react-selectors";
+import {navigateToAppliancePage, rescanFromAppliancePage} from "./helpers/navigation";
+import {parseLocalStorage} from "./helpers/localStorage";
+import translation from "../src/app/locales/translation";
+const exec = require("child_process").exec;
+
+// TODO: Find a way how to import that
+const MAX_SELECT_APPLIANCES = 4;
+const MAX_APPLIANCES_ON_PAGE = 5;
+
+const UNCONFIGURED_SCREEN = false;
+const CONFIGURED_SCREEN = true;
+
+const CMD_APPLIANCE = "PSA_--ApplianceTest--_v1_X_EX-1_HWType_Block_0";
+const CMD_CLUSTER = "PSC_--ClusterTest--_v1_X_EX-1_HWType_Block_3";
+
+fixture("AppliancePage container tests")
+    .page("../src/index.html")
+    .beforeEach(async () => {
+        await waitForReact();
+    });
+
+/**
+ * Test the pagination buttons on Appliances page
+ * @param {Boolean} isConfiguredScreen - set the proper screen for testing
+ */
+const testPaginationButtons = async isConfiguredScreen => {
+    await navigateToAppliancePage(isConfiguredScreen);
+
+    const applianceList = await parseLocalStorage(isConfiguredScreen);
+
+    const pagination = Math.ceil(applianceList.length / MAX_APPLIANCES_ON_PAGE);
+
+    let paginationButton;
+    let actualTopApplianceName;
+    let calculatedTopApplianceName;
+
+    for (let i = 1; i < pagination + 1; i++) {
+        // Choose a new button in pagination
+        paginationButton = ReactSelector("li")
+            .withKey(i)
+            .find("a");
+
+        await t.expect(paginationButton.exists).ok();
+        await t.click(paginationButton);
+
+        // Compare top appliance name from the page and from the appliance list
+        actualTopApplianceName = ReactSelector("Appliance")
+            .nth(0)
+            .find("p.app-name");
+        calculatedTopApplianceName = applianceList[(i - 1) * MAX_APPLIANCES_ON_PAGE].name;
+
+        await t.expect(actualTopApplianceName.textContent).eql(calculatedTopApplianceName);
+    }
+};
+
+/**
+ * Test the ability to detect new appliances & clusters
+ * @param {string} cmdArg
+ * @param {Boolean} isConfiguredScreen
+ */
+const testDetection = async (cmdArg, isConfiguredScreen) => {
+    const platform = process.platform;
+
+    await navigateToAppliancePage(isConfiguredScreen);
+
+    let firstAppliance = ReactSelector("Appliance")
+        .nth(0)
+        .find("p.app-name");
+    const testName = cmdArg.split("_")[1];
+
+    let publish;
+
+    // Check that test appliance hasn't been published yet
+    await t.expect(firstAppliance.textContent).notEql(testName);
+
+    // Execute publishing script
+    if (platform === "win32") {
+        publish = exec(`dns-sd -R ${cmdArg} _http._tcp . 3000`);
+    } else if (platform === "linux") {
+        publish = exec(`avahi-publish-service ${cmdArg} _http._tcp 3000`);
+    } else {
+        console.log("This test is not runnable in your OS");
+        return;
```

Comment:
The test will not be skipped I guess since you return from the `testDetection`, not from the `test` function.